driver/src/order_driver.rs

```diff
@@ -0,0 +1,78 @@
+use crate::contract::SnappContract;
+use crate::db_interface::DbInterface;
+use crate::error::{DriverError, ErrorKind};
+use crate::models::{RollingHashable, Serializable};
+use crate::models;
+use crate::price_finding::{PriceFinding, Solution};
+use crate::util;
+
+use web3::types::{H256, U256};
+
+pub fn run_order_listener<D, C, PF>(
+    db: &D, 
+    contract: &C, 
+    price_finder: &mut PF
+) -> Result<bool, DriverError>
+    where   D: DbInterface,
+            C: SnappContract,
+            PF: PriceFinding
+{
+    let auction_slot = contract.get_current_auction_slot()?;
+
+    println!("Current top auction slot is {:?}", auction_slot);
+    let slot = util::find_first_unapplied_slot(
+        auction_slot + 1, 
+        Box::new(&|i| contract.has_auction_slot_been_applied(i))
+    )?;
+    if slot <= auction_slot {
+        println!("Highest unprocessed auction slot is {:?}", slot);
+        if util::can_process(slot, contract,
```

Comment:
here, we might want to extend the  function can_process with an additional parametric function: slot_is_full: slot_nr => contract.batch_already_full()
which would check for deposits the nr of deposits and for orders the nr of orders and if they already reached the threshold, it can return true. 

I think it is important to accelerate processing.

Reply:
Sounds good. Can you submit a separate PR?